tests/plugins/test_sendmail.py

```diff
@@ -294,6 +320,11 @@ class WF(object):
         }
 
         workflow = WF()
+        if annotations and metadata_stored:
+            mock_store_metadata_results(workflow)
+        if not annotations and metadata_stored:
+            mock_store_missing_metadata_results(workflow)
```

Comment:
Does `annotations` make any sense if `metadata_stored` evaluates as false? If not, I'd rather see the test the other way around, e.g.:
```
if metadata_stored:
    if annotations:
        mock_store_metadata_results(workflow)
    else:
        mock_store_missing_metadata_results(workflow)
```